projects/igniteui-angular/src/lib/icon/icon.component.ts

```diff
@@ -226,21 +231,21 @@ export class IgxIconComponent implements OnInit, OnDestroy {
     }
 
     /**
-     *  An accessor that returns the key of the SVG image.
-     *  The key consists of the font-family and the name separated by underscore.
+     *  An accessor that returns the underlying SVG image as SafeHtml.
      *
      * @example
      * ```typescript
      * @ViewChild("MyIcon")
      * public icon: IgxIconComponent;
      * ngAfterViewInit() {
-     *    let svgKey = this.icon.getSvgKey;
+     *    let svg: SafeHtml = this.icon.getSvg;
      * }
      * ```
      */
-    public get getSvgKey(): string {
+    public get getSvg(): SafeHtml {
         if (this.iconService.isSvgIconCached(this.name, this.family)) {
-            return '#' + this.iconService.getSvgIconKey(this.name, this.family);
+            const svgText = this.iconService.getSvgIcon(this.name, this.family);
+            return this.sanitizer.bypassSecurityTrustHtml(svgText);
```

Comment:
I'm not completely sure about this. But I've experienced excesive DOM refresh issues with getter + sanitizer.
The issue is that the sanitizer returns a brand new object each time which is not strictly equal to the previous one and Angular keeps refreshing it.
What I've done is cache the sanitized result. In this case I would keep: `lastIconKey: this.name + "/" + this.family;` and `lastIconContent = sanitizedString;`
Then compare if the key has changed and if not return the lastIconContent.